lib/plug/crypto/message_encryptor.ex

```diff
@@ -175,37 +134,18 @@ defmodule Plug.Crypto.MessageEncryptor do
         :error
     end
     |> case do
-      cek when bit_size(cek) in [128, 192, 256] ->
-        {:ok, cek}
-
-      _ ->
-        :error
+      cek when bit_size(cek) in [128, 192, 256] -> {:ok, cek}
+      _ -> :error
     end
   end
 
-  defp encode_token(protected, encrypted_key, iv, cipher_text, cipher_tag) do
-    Base.url_encode64(protected, padding: false)
-    |> Kernel.<>(".")
-    |> Kernel.<>(Base.url_encode64(encrypted_key, padding: false))
+  defp encode_token(head, iv, cipher_text, cipher_tag) do
```

Comment:
(nit) The encoded format could be simplified for "v2" to the following since everything should have a fixed size (for XChaCha20-Poly1305):

```elixir
version = 2
raw = <<version::8, iv::bits-size(192), cipher_tag::bits-size(128), cipher_text::bytes>>
Base.urlsafe_encode64(raw)
```

Decoding for decryption could then look like this:

```elixir
case :binary.split(encrypted, ".", [:global]) do
  [encoded] ->
    case Base.urlsafe_decode64(encoded, padding: false) do
      <<2::8, iv::bits-size(192), cipher_tag::bits-size(128), cipher_text::bytes>> ->
        # XChaCha20-Poly1305 ...
      
      <<3::8, ...>> ->
        # some future version ...

    end

  [protected, encrypted_key, iv, cipher_text, cipher_tag] ->
    # AES-128-GCM backwards compatibility ...

end
```

Reply:
Great idea, I have incorporated this but still kept the first 4 digits outside of base64. :)